lib/index.js

```diff
@@ -15,151 +15,184 @@ const Manager = require('./manager');
-    return this.response(manager._response(template, context, options, this.request));
-};
+        const self = this;
 
+        return function (template, context, options) {
 
-internals.handler = function (route, options) {
+            const manager = self.nearestManager(this.realm);
 
-    Joi.assert(options, Schemas.handler, 'Invalid view handler options (' + route.path + ')');
+            Hoek.assert(manager, 'Missing views manager');
 
-    if (typeof options === 'string') {
-        options = { template: options };
+            return this.response(manager._response(template, context, options, this.request));
+        };
     }
 
-    const settings = {                                                // Shallow copy to allow making dynamic changes to context
-        template: options.template,
-        context: options.context,
-        options: options.options
-    };
 
-    return function (request, h) {
+    handler(rootServer) {
 
-        const context = {
-            params: request.params,
-            payload: request.payload,
-            query: request.query,
-            pre: request.pre
-        };
+        return function (route, options) {
+
+            Joi.assert(options, Schemas.handler, 'Invalid view handler options (' + route.path + ')');
 
-        if (settings.context) {                                     // Shallow copy to avoid cloning unknown objects
-            const keys = Object.keys(settings.context);
-            for (let i = 0; i < keys.length; ++i) {
-                const key = keys[i];
-                context[key] = settings.context[key];
+            if (typeof options === 'string') {
+                options = { template: options };
             }
-        }
 
-        return h.view(settings.template, context, settings.options);
-    };
-};
+            const settings = {                                                // Shallow copy to allow making dynamic changes to context
+                template: options.template,
+                context: options.context,
+                options: options.options
+            };
+
+            return function (request, h) {
+
+                const context = {
+                    params: request.params,
+                    payload: request.payload,
+                    query: request.query,
+                    pre: request.pre
+                };
+
+                if (settings.context) {                                     // Shallow copy to avoid cloning unknown objects
+                    const keys = Object.keys(settings.context);
+                    for (let i = 0; i < keys.length; ++i) {
+                        const key = keys[i];
+                        context[key] = settings.context[key];
+                    }
+                }
+
+                return h.view(settings.template, context, settings.options);
+            };
+        };
+    }
 
 
-internals.state = (realm) => {
+    getState(realm) {
```

Comment:
I find it a little funny having instance methods that don't utilize `this`.  Any reason this helper is moved out of `internals`?